concepts/05 Widgets/TagBox/12 Limit the Tag Count/10 Selected Tags.md

```diff
@@ -86,5 +86,141 @@ The following code shows the [onValueChanged](/api-reference/10%20UI%20Widgets/d
          // ...
      })
 
+##### Vue
+
+    <template>
+        <div>
+            <DxTagBox
+                id="tagBoxContainer"
+                :data-source="products"
+                @value-changed="onValueChanged"
+            />
+            <DxTooltip 
+                target="#tagBoxContainer"
+                :hide-event="hideEvent"
+                position="bottom"
+                :visible="isTooltipVisible">
+                <template>
+                    <p>Limit reached</p>
+                </template>
+            </DxTooltip>
+        </div>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { DxTagBox } from 'devextreme-vue/tag-box';
+    import { DxTooltip } from 'devextreme-vue/tooltip';
+
+    export default {
+        components: {
+            DxTagBox,
+            DxTooltip
+        },
+        data() {
+            return {
+                hideEvent: { name: 'mouseout', delay: 2000 },
+                products: [
+                    "HD Video Player",
+                    "SuperHD Video Player",
+                    "SuperPlasma 50",
+                    "SuperLED 50",
+                    // ...
+                ],
+                isTooltipVisible: false,
+                maxItems: 2
+            };
+        },
+        methods: {
+            onValueChanged(e) {
+                if (e.value.length > this.maxItems) {
+                    const newValue = e.value.slice(0, this.maxItems);
+                    e.component.option('value', newValue);
+                    this.isTooltipVisible = true;
+                }
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { TagBox } from 'devextreme-react/tag-box';
+    import { Tooltip } from 'devextreme-react/tooltip';
+
+    const products = [
+        "HD Video Player",
+        "SuperHD Video Player",
+        "SuperPlasma 50",
+        "SuperLED 50",
+        // ...
+    ];
+
+    const renderContent = () => {
+        return (
+            <p>Limit reached</p>
+        );
+    }
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                isTooltipVisible: false
+            }
+
+            this.maxItems = 2;
```

Comment:
Should we declare it as a constant outside of the class?

Reply:
Not necessarily.